tools/ivwpy/regression/reporttest.py

```diff
@@ -55,30 +53,42 @@ def __init__(self, key, differenceTolerance=0.0):
         self.differenceTolerance = differenceTolerance
 
     def test(self, report):
-        imgs = report[self.key]
+        imgs = report[self.key]["tests"]
         for img in imgs:
             tol = safeget(report, "config", "image_test", "differenceTolerance",
                           img["image"], failure=self.differenceTolerance)
             tol = max(tol, self.differenceTolerance)
 
             if img['test_mode'] != img['ref_mode']:
                 self.message[img['image']] = \
-                    ("Image {image} has different modes, " +
+                    ("Image {image} has different modes, "
                      "Test: {test_mode} vs Reference:{ref_mode}").format(**img)
             elif img['test_size'] != img['ref_size']:
                 self.message[img['image']] = \
-                    ("Image {image} has different sizes, " +
+                    ("Image {image} has different sizes, "
                      "Test: {test_size} vs Reference:{ref_size}").format(**img)
             elif img["difference"] > tol:
                 self.message[img['image']] = \
-                    ("Image {image} has difference greater then the allowd tolerance ({difference}% &gt; {tol})  " +
-                     "difference, {different_pixels} different pixels, " +
+                    ("Image {image} has difference greater then the allowed  "
+                     "tolerance ({difference}% &gt; {tol})  "
+                     "difference, {different_pixels} different pixels, "
                      "largest difference {max_difference}").format(tol=tol, **img)
 
         return len(self.message) == 0
 
-    def failures(self):
-        return {self.key: self.message}
+
+class ReportTextTest(ReportTest):
+    def __init__(self, key):
+        self.key = key
+        self.message = {}
+
+    def test(self, report):
+        tests = report[self.key]["tests"]
+        for test in tests:
+            if test['diff'] > 0:
+                self.message[test['txt']] = f"{test['txt']} differs from the reference" \
```

Comment:
why `txt` ? Were there not also csv files?

Reply:
txt mean any text based file, xml, csv, txt, anythnig that you can diff with a textual diff more or less